app/src/main/java/com/github/dedis/student20_pop/utility/protocol/LowLevelProxy.java

```diff
@@ -0,0 +1,59 @@
+package com.github.dedis.student20_pop.utility.protocol;
+
+import com.github.dedis.student20_pop.model.network.level.high.Data;
+
+import java.io.Closeable;
+import java.util.List;
+import java.util.concurrent.CompletableFuture;
+
+public interface LowLevelProxy extends Closeable {
+
+    /**
+     * Minimum number of millisecond a request can stay unanswered before it is classified as timeout
+     */
+    long REQUEST_TIMEOUT = 5000; // 5s
+
+    /**
+     * Subscribe to a channel
+     *
+     * @param channel to subscribe to
+     * @return a completable future holding the response value
+     */
+    CompletableFuture<Integer> subscribe(String channel);
+
+    /**
+     * Unsubscribe to a channel
+     *
+     * @param channel to subscribe to
+     * @return a completable future holding the response value
+     */
+    CompletableFuture<Integer> unsubscribe(String channel);
+
+    /**
+     * Publish an event on given channel
+     *
+     * @param channel to publish the event on
```

Comment:
The comments for the parameters `sender` and `key` are missing

Reply:
Added in latest commit